task_6_1.py

```diff
@@ -0,0 +1,59 @@
+import sys
+
+
+def validate_number(number_string):
+    try:
+        float(number_string)
+        return True
+    except ValueError:
+        return False
+
+
+def validate_input(input_lines):
+    """Check that the list of input_lines is a valid list of numbers separated by a newline character"""
+    for the_line in input_lines:
+        if the_line.strip() == '':
+            continue
+        if not validate_number(the_line.strip()):
+            return False
+    return True
+
+
+def validate_input_file(lines):
+    if not validate_input(lines):
+        print('Invalid input file. Numbers must start each time on a new line.')
+        sys.exit()
+
+
+def open_to_read():
+    # open the file with numbers for reading
+    try:
+        with open('numbers.txt', 'r', encoding='utf-8') as numbers_file:
+            # read all lines of the file and convert them into valid numbers
+            lines = [line.strip() for line in numbers_file.readlines()]
+            validate_input_file(lines)  #
```

Comment:
rewrite your function **validate_input_file** for work with single line. Then you could use it in comprehension
```suggestion
            lines = [line.strip() for line in numbers_file.readlines() if validate_input_file(line)]
```